### PR TITLE
Add Deployment of ironic-neutron-agent

### PR DESCRIPTION
**Pull request by hjensas**
In ironic controller add deployment of ironic neutron agent. When replicas is ! > 0 the ironic neutron agent will not be deployed, setting replicas = 0 will also delete any running deployment.

Also:
* Update sample files to use zed container images for ironic neutron agent ... match the other components.
* Update description string for inspector

Jira: [OSP-24274](https://issues.redhat.com//browse/OSP-24274)

### CODE REVIEW

File: config/samples/ironic_v1beta1_ironic.yaml [hunk 1]
```diff
@@ -22,4 +22,7 @@ spec:
     replicas: 1
     containerImage: quay.io/podified-antelope-centos9/openstack-ironic-inspector:current-podified
     pxeContainerImage: quay.io/podified-antelope-centos9/openstack-ironic-pxe:current-podified
-  secret: osp-secret
+  secret: ironic-secret
```
Comment (steveb) [inline config/samples/ironic_v1beta1_ironic.yaml:25]: is ironic-secret only for standalone? So this should remain osp-secret?

**Review by steveb — COMMENTED** [review]
Looks good, just one query about the samples secret

File: config/samples/ironic_v1beta1_ironic.yaml (hunk 1 above)
Comment (hjensas) [reply config/samples/ironic_v1beta1_ironic.yaml:25]: ooops, that looks like a rebase conflict resolving mistake on my part.
I will update.

**Review by steveb — APPROVED** [review]
/lgtm

**Pull request merged**